java/vector/src/main/java/org/apache/arrow/vector/ipc/message/ArrowBatchMessage.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.vector.ipc.message;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Wrapper around flatbuffer message table.  This keeps from exposing
+ * flatbuffer details to upstream packages.
+ */
+public class ArrowBatchMessage {
```

Comment:
is the name `ArrowBatchMessage` misleading because this can wrap all `Message`s not just `ArrowRecordBatch` or `ArrowDictionaryBatch` right?

Reply:
renamed to ArrowMessageHeader

Reply:
Actually that is always something in the flatbuffers schema.  let me pick something different.

Reply:
renamed to ArrowMessageMetadata

Reply:
It turns out this duplicate most of the functionality on an already existing class.  I've removed this and added functionality to the new class.